### PR TITLE
Add some Proton per-game workarounds

### PR DESCRIPTION
**Pull request by Joshua-Ashton**

### CODE REVIEW

File: dxvk.conf [hunk 1]
```diff
@@ -31,6 +31,15 @@
 # d3d9.customVendorId = 0000
 
 
+# Override the reported device description
+#
+# Supported values: Any string.
+
+# dxgi.customDeviceDesc = ""
+
```
Comment (doitsujin) [inline dxvk.conf:39]: no space here please

File: src/dxgi/dxgi_adapter.cpp [hunk 2]
```diff
@@ -283,6 +283,15 @@ namespace dxvk {
...
         sharedMemory += heap.size;
     }
+
+    // Some games think we are on Intel given a lack of
+    // NVAPI or AGS/atiadlxx support.
+    // Report our device memory as shared memory,
+    // and some small amount for the carveout.
+    if (options->emulateUMA && !m_adapter->isUnifiedMemoryArchitecture()) {
+      deviceMemory = 128 * (1 << 20);
+      sharedMemory = deviceMemory;
```
Comment (doitsujin) [inline src/dxgi/dxgi_adapter.cpp:293]: surely this should be the other way around? We're now reporting 128MB for *both* shared and device memory.

**Review by doitsujin — CHANGES_REQUESTED** [review]

File: src/dxgi/dxgi_adapter.cpp (hunk 2 above)
Comment (Joshua-Ashton) [reply src/dxgi/dxgi_adapter.cpp:293]: Yes this should :frog:

File: src/dxgi/dxgi_adapter.cpp [hunk 3]
```diff
@@ -278,6 +283,15 @@ namespace dxvk {
       else
         sharedMemory += heap.size;
     }
+
+    // Some games think we are on Intel given a lack of
+    // NVAPI or AGS/atiadlxx support.
+    // Report our device memory as shared memory,
+    // and some small amount for the carveout.
+    if (options->emulateUMA && !m_adapter->isUnifiedMemoryArchitecture()) {
+      sharedMemory = deviceMemory;
```
Comment (DadSchoorse) [inline src/dxgi/dxgi_adapter.cpp:292]: Won't this be bad on low VRAM cards, like 2GB?

Comment (doitsujin) [reply src/dxgi/dxgi_adapter.cpp:292]: i mean this is literally a GTA IV hack, whatever makes that game happy should work here. Most games seem to ignore this value anyway.

**Pull request merged**